firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -198,9 +200,13 @@ class TabManagerImplementation: NSObject, TabManager, FeatureFlaggable {
         if #available(iOS 15.4, *) {
             configuration.preferences.isElementFullscreenEnabled = true
         }
+
+        // Use shared pool for regular mode, isolated pool for private mode
         if isPrivate {
+            configuration.processPool = WKProcessPool()
```

Comment:
I think we should share the process pool between private tabs as well. I asked around and took some notes [here](https://mozilla-hub.atlassian.net/browse/FXIOS-11896?focusedCommentId=1042122).

I also wonder looking at the changes here if it would be just simpler to start using the [`DefaultWKEngineConfigurationProvider`](https://github.com/mozilla-mobile/firefox-ios/blob/main/BrowserKit/Sources/WebEngine/WKWebview/WKEngineConfigurationProvider.swift) already instead. We have a task for this in the tab tray refactor epic, and I worry we make changes here that we'll be re-changing in a couple of weeks anyway and could introduce regressions again. I would like us to avoid back and forth on this matter if possible